### DISCUSSION TITLE
Can build but cannot upload

### DISCUSSION THREAD

**Discussion by Trev114** [PlummersSoftwareLLC/NightDriverStrip#155] [Q&A]
I can get PlatformIO to build but when it comes time to upload it fails every time. I am working on a windows machine.
Here is what I get:

 *  Executing task: C:\Users\trev1\.platformio\penv\Scripts\platformio.exe run --target upload 

Processing mesmerizer (board: esp-wrover-kit; platform: espressif32; framework: arduino)
-------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp-wrover-kit.html
PLATFORM: Espressif 32 (5.2.0) > Espressif ESP-WROVER-KIT
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (ftdi) On-board (ftdi) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.20005.220925 (2.0.5)
 - tool-esptoolpy @ 1.40201.0 (4.2.1)
 - tool-mkfatfs @ 2.0.1
 - tool-mklittlefs @ 1.203.210628 (2.3)
 - tool-mkspiffs @ 2.230.0 (2.30)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch3
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 48 compatible libraries
Scanning dependencies...
Dependency Graph
|-- IRremoteESP8266 @ 2.8.4
|-- FastLED @ 3.5.0
|   |-- SmartMatrix @ 4.0.3+sha.8e625f8
|   |   |-- SPI @ 2.0.0
|   |   |-- Adafruit GFX Library @ 1.11.3
|   |   |   |-- Adafruit BusIO @ 1.14.1
|   |   |   |   |-- Wire @ 2.0.0
|   |   |   |   |-- SPI @ 2.0.0
|   |   |   |-- Wire @ 2.0.0
|   |   |   |-- SPI @ 2.0.0
|   |-- SPI @ 2.0.0
|-- Adafruit BusIO @ 1.14.1
|   |-- Wire @ 2.0.0
|   |-- SPI @ 2.0.0
|-- Adafruit GFX Library @ 1.11.3
|   |-- Adafruit BusIO @ 1.14.1
|   |   |-- Wire @ 2.0.0
|   |   |-- SPI @ 2.0.0
|   |-- Wire @ 2.0.0
|   |-- SPI @ 2.0.0
|-- Adafruit ILI9341 @ 1.5.12
|   |-- Adafruit GFX Library @ 1.11.3
|   |   |-- Adafruit BusIO @ 1.14.1
|   |   |   |-- Wire @ 2.0.0
|   |   |   |-- SPI @ 2.0.0
|   |   |-- Wire @ 2.0.0
|   |   |-- SPI @ 2.0.0
|   |-- Adafruit BusIO @ 1.14.1
|   |   |-- Wire @ 2.0.0
|   |   |-- SPI @ 2.0.0
|   |-- Wire @ 2.0.0
|   |-- SPI @ 2.0.0
|-- U8g2 @ 2.34.5
|   |-- SPI @ 2.0.0
|   |-- Wire @ 2.0.0
|-- arduinoFFT @ 1.5.6
|-- AsyncTCP @ 1.1.1
|-- ESP Async WebServer @ 1.2.3+sha.a6cd518
|   |-- AsyncTCP @ 1.1.1
|   |-- ArduinoJson @ 6.19.4
|   |-- FS @ 2.0.0
|   |-- WiFi @ 2.0.0
|-- ArduinoJson @ 6.19.4
|-- Bounce2 @ 2.71.0
|-- RemoteDebug @ 3.0.5+sha.706f093
|   |-- WiFi @ 2.0.0
|   |-- SPI @ 2.0.0
|   |-- WiFiClientSecure @ 2.0.0
|   |   |-- WiFi @ 2.0.0
|-- QRCode @ 0.0.1
|-- SmartMatrix @ 4.0.3+sha.8e625f8
|   |-- SPI @ 2.0.0
|   |-- Adafruit GFX Library @ 1.11.3
|   |   |-- Adafruit BusIO @ 1.14.1
|   |   |   |-- Wire @ 2.0.0
|   |   |   |-- SPI @ 2.0.0
|   |   |-- Wire @ 2.0.0
|   |   |-- SPI @ 2.0.0
|-- ArduinoOTA @ 2.0.0
|   |-- Update @ 2.0.0
|   |-- WiFi @ 2.0.0
|   |-- ESPmDNS @ 2.0.0
|   |   |-- WiFi @ 2.0.0
|-- HTTPClient @ 2.0.0
|   |-- WiFi @ 2.0.0
|   |-- WiFiClientSecure @ 2.0.0
|   |   |-- WiFi @ 2.0.0
|-- WiFiClientSecure @ 2.0.0
|   |-- WiFi @ 2.0.0
|-- SPI @ 2.0.0
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- SPIFFS @ 2.0.0
|   |-- FS @ 2.0.0
Building in debug mode
Retrieving maximum program size .pio\build\mesmerizer\firmware.elf
Checking size .pio\build\mesmerizer\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [==        ]  19.8% (used 64888 bytes from 327680 bytes)
Flash: [==========]  96.0% (used 1383401 bytes from 1441792 bytes)
Configuring upload protocol...
AVAILABLE: cmsis-dap, esp-bridge, esp-prog, espota, esptool, ftdi, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa
CURRENT: upload_protocol = esptool
Looking for upload port...
Using manually specified: COM3
Uploading .pio\build\mesmerizer\firmware.bin
esptool.py v4.2.1
Serial port COM3
Connecting......
Chip is ESP32-D0WDQ6 (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Traceback (most recent call last):
WARNING: Detected crystal freq 27.07MHz is quite different to normalized freq 26MHz. Unsupported crystal in use?   
Crystal is 26MHz
  File "C:\Users\trev1\.platformio\packages\tool-esptoolpy\esptool.py", line 34, in <module>
MAC: 30:ae:a4:bb:54:80
    esptool._main()
Uploading stub...
  File "C:\Users\trev1\.platformio\packages\tool-esptoolpy\esptool\__init__.py", line 1004, in _main
Running stub...
    main()
Stub running...
  File "C:\Users\trev1\.platformio\packages\tool-esptoolpy\esptool\__init__.py", line 790, in main
Changing baud rate to 200000    esp.flash_set_parameters(flash_size_bytes(args.flash_size))
0
Changed.
  File "C:\Users\trev1\.platformio\packages\tool-esptoolpy\esptool\loader.py", line 1108, in flash_set_parameters  
Configuring flash size...
    self.check_command(
  File "C:\Users\trev1\.platformio\packages\tool-esptoolpy\esptool\loader.py", line 406, in check_command
    val, data = self.command(op, data, chk, timeout=timeout)
  File "C:\Users\trev1\.platformio\packages\tool-esptoolpy\esptool\loader.py", line 375, in command
    p = self.read()
  File "C:\Users\trev1\.platformio\packages\tool-esptoolpy\esptool\loader.py", line 307, in read
    return next(self._slip_reader)
StopIteration
*** [upload] Error 1
=========================================== [FAILED] Took 35.45 seconds ===========================================
Environment    Status    Duration
-------------  --------  ------------
mesmerizer     FAILED    00:00:35.451
====================================== 1 failed, 0 succeeded in 00:00:35.451 ======================================
 *  The terminal process "C:\Users\trev1\.platformio\penv\Scripts\platformio.exe 'run', '--target', 'upload'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.

**Comment by Trev114** [comment]
I figured out the issue. Forgive me Im very new to this stuff

**Reply by rbergen** [reply]
No worries, everybody was new to this at some point.
Just in case someone else runs into the same problem, would you mind adding a quick explanation of what the problem was, and what you did to solve it?

**Comment by Trev114** [comment]
To solve this issue I went to "PlatformIO.ini" and noticed that "default_envs" was set to "mesmerizer" which is a different board than what I was using. I scrolled through the list to find my board which was labeled "led strip". I changed "mesmerizer" to "led strip" then built the file and uploaded without an issue.

**Reply by CdnLaser** [reply]
How or better yet, where did you find the info to change the 'default_envs' ?? I have a LilyGo esp32-c3 lcd version and dont know where to get the info to insert in  the var. I tried it as a 'Pico' because that is what showed up on the lcd when it booted initially, but the com port is erroring out .

**Reply by Trev114** [reply]
I watched Daves video of the leds with the presents. I think it was the last LED one he uploaded. And saw it on there. I couldnt get the code to work either with keeping the upload and monitor ports as were in the file. I changed my to COM3 because that is the serial port my heltec was on per my device manager.
